demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -41,26 +57,36 @@
  * This demo uses the Mosquitto test server. This is a public MQTT server; do not
  * publish anything sensitive to this server.
  */
-#define SERVER                 "test.mosquitto.org"
+#define BROKER_ENDPOINT           "test.mosquitto.org"
+
+/**
+ * @brief Length of MQTT server host name.
+ */
+#define BROKER_ENDPOINT_LENGTH    ( ( uint16_t ) ( sizeof( BROKER_ENDPOINT ) - 1 ) )
 
 /**
  * @brief MQTT server port number.
  *
  * In general, port 8883 is for secured MQTT connections.
  */
-#define PORT                   8883
+#define BROKER_PORT               ( 8883 )
 
 /**
  * @brief Path of the file containing the server's root CA certificate.
  *
  * This certificate should be PEM-encoded.
  */
-#define SERVER_CERT            "mosquitto.org.crt"
+#define SERVER_CERT_PATH          "certificates/mosquitto.org.crt"
```

Comment:
Does this demo only run with mosquitto? Could this be changed to another cert at compile time with a cmake flag?

Reply:
Maybe we can copy all *.crt or *.pem into the build/bin/certificates directory. The cmake flag works too though, but we would need to document that for the user.

Reply:
Actually, I'm not sure if we should open the certificate using `fopen`. The relative directory is based on the location of the current working directory rather than the location of the executable. We could `chdir()` to the path of the executable, but the code we write to find that path might not always work depending on the OS. Maybe we should just place the certificate in the header file like in AFR.

Reply:
I like that proposal. That way we will be consistent in server auth and mutual auth demo. Will update the PR to do that. Thanks

Reply:
Thinking about it again, I am keeping it this way because the mosquitto server certificate is downloadable as a file and a small transformation of that is needed to keep it as macro. File will be easier for an end customer.

Reply:
We can specify path in demo_config.h ( like we do in iot_config.h) in v4_beta.